src/core/scannedfoldersmodel.cpp

```diff
@@ -93,27 +109,60 @@ QVariant ScanFoldersModel::headerData(int section, Qt::Orientation orientation,
   if (orientation != Qt::Horizontal || role != Qt::DisplayRole || section < 0 || section >= columnCount())
     return QVariant();
 
-  if (section == PathColumn)
-    return tr("Watched Folder");
-  return tr("Download here");
+  switch (section)  {
+      case PathColumn: return tr("Watched Folder");
+      case DownloadAtTorrentColumn: return tr("Download here");
+      case DownloadPath: return tr("Download path");
+  }
+
+   return QVariant();
 }
 
 Qt::ItemFlags ScanFoldersModel::flags(const QModelIndex &index) const {
-  if (!index.isValid() || index.row() >= rowCount() || index.column() != DownloadAtTorrentColumn)
-    return QAbstractTableModel::flags(index);
-  return QAbstractTableModel::flags(index) | Qt::ItemIsUserCheckable;
+  if (!index.isValid() || index.row() >= rowCount())
+      return QAbstractTableModel::flags(index);
+
+  const PathData* pathData = m_pathList.at(index.row());
+
+  switch (index.column())  {
+  case PathColumn:                  return QAbstractTableModel::flags(index);
+  case DownloadAtTorrentColumn:     return QAbstractTableModel::flags(index) | Qt::ItemIsUserCheckable;
+  case DownloadPath:    {
+          if (pathData->downloadAtPath == false){
+            return QAbstractTableModel::flags(index) | Qt::ItemIsEditable | Qt::ItemIsEnabled;
+          }
+          else{   //dont edit DownloadPath if checked 'downloadAtPath'
+            return QAbstractTableModel::flags(index) ^ Qt::ItemIsEnabled;
+          }
+      }
+  }
+  return QAbstractTableModel::flags(index);
```

Comment:
just leave an empty line before this return to visually separate it from the switch()